diff --git a/dist/core/index.mjs b/dist/core/index.mjs
index c485b314ccb2426f5d7fc7fadd28199714db6420..2cd1f627b68aea91ba162dfad324306a95e5c772 100644
--- a/dist/core/index.mjs
+++ b/dist/core/index.mjs
@@ -1682,8 +1682,9 @@ async function compressPublicAssets(nitro) {
     dot: true,
     ignore: ["**/*.gz", "**/*.br"]
   });
-  await Promise.all(
-    publicFiles.map(async (fileName) => {
+  await runParallel(
+    new Set(publicFiles),
+    async (fileName) => {
       const filePath = resolve(nitro.options.output.publicDir, fileName);
       if (existsSync(filePath + ".gz") || existsSync(filePath + ".br")) {
         return;
@@ -1724,7 +1725,8 @@ async function compressPublicAssets(nitro) {
           await fsp.writeFile(compressedPath, compressedBuff);
         })
       );
-    })
+    },
+    { concurrency: 10 }
   );
 }
 function isTextMime(mimeType) {
